vnext/ReactUWP/Utils/Helpers.cpp

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "Helpers.h"
+#include <Modules/NativeUIManager.h>
+
+namespace react { namespace uwp {
+
+ReactId getReactId(IReactInstance *instance, winrt::FrameworkElement const& fe)
+{
+  ReactId reactId;
+  assert(instance);
```

Comment:
>assert(instance); [](start = 2, length = 17)

I would prefer we use SAL annotations (in this case _In_) to asserting required parameters.

Reply:
Can we use SAL in this repo?

My main concern though is that SAL is ton of work and I rarely see it pay off because it never gets used anything (and it's not useful for documentation purposes since it is so cryptic). Does DEP have systems that make use of the info? I realize this conversation is probably outside the scope of this review!

---
In reply to: [295456661](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295456661) [](ancestors = 295456661)

Reply:
Yes SAL works (at least the small test I did works).  DEP makes use of it through OACR.  I agree it can be cryptic, but it seems useful for the most basic cases like _In_ and _In_opt_.

We need to have the broader conversation about coding standards in this code base - right now it's a mix of styles, some of which clash with the standards here in Windows.

---
In reply to: [295526981](https://github.com/microsoft/react-native-windows/pull/2623#discussion_r295526981) [](ancestors = 295526981,295456661)

Reply:
Done